crates/primitives/starknet/src/transaction/types.rs

```diff
@@ -372,6 +387,114 @@ impl From<DeployAccountTransaction> for Transaction {
         }
     }
 }
+
+/// Error of conversion between the Madara Primitive Transaction and the RPC Transaction
+#[cfg(feature = "std")]
+#[derive(Debug)]
+pub enum RPCTransactionConversionError {
+    /// The u8 stored version doesn't match any of the existing version at the RPC level
+    UnknownVersion,
+    /// Missing information
+    MissingInformation,
+    /// Conversion from byte array has failed.
+    FromArrayError,
+    /// Provided byte array has incorrect lengths.
+    InvalidLength,
+    /// Invalid character in hex string.
+    InvalidCharacter,
+    /// Value is too large for FieldElement (felt252).
+    OutOfRange,
+}
+
+#[cfg(feature = "std")]
+impl From<Felt252WrapperError> for RPCTransactionConversionError {
+    fn from(value: Felt252WrapperError) -> Self {
+        match value {
+            Felt252WrapperError::FromArrayError => Self::FromArrayError,
+            Felt252WrapperError::InvalidLength => Self::InvalidLength,
+            Felt252WrapperError::InvalidCharacter => Self::InvalidCharacter,
+            Felt252WrapperError::OutOfRange => Self::OutOfRange,
+        }
+    }
+}
+
+#[cfg(feature = "std")]
+impl TryFrom<Transaction> for RPCTransaction {
+    type Error = RPCTransactionConversionError;
+    fn try_from(value: Transaction) -> Result<Self, Self::Error> {
+        let transaction_hash = value.hash.0;
```

Comment:
For all this section, I would recommend to abstract details implementation. Doing `.0` is too much associated to the `Felt252Wrapper` implem.

This can be more idiomatic, and more robust to future changes:
`let transaction_hash: FieldElement = value.hash.into();`

This comment is also for all lines below, where `.0` is used.

Reply:
actually I hesitated but I eventually chose the `.0` instead to especially avoid to explicitly import and cast to `FieldElement`. I've guessed that a tuple type called `FeltWrapper` with one single `pub` item may actually change its internal felt type, but not the fact that it's at index 0.

Of course ready to change to whatever is recommended! But I'd like to have some more views on this because with the same concern I came to the opposite conclusion!

Reply:
I tend to slightly agree with @glihm here. But I understand your argument @ClementWalter . For the sake of closing RPC endpoints ASAP I am fine merging as is.